src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.cs

```diff
@@ -136,7 +136,9 @@ void Complete(ContentDialogClosingEventArgs args)
 					_popup.Child = null;
 					UpdateVisualState();
 					Closed?.Invoke(this, new ContentDialogClosedEventArgs(result));
-					_tcs.SetResult(result);
+
+					(var tcs, _tcs) = (_tcs, null);
+					DispatcherQueue.TryEnqueue(() => tcs?.TrySetResult(result));
```

Comment:
Hum is this really needed? It's the responsibility of the awaiter to go back on the UIThread, not the caller 🤔

Reply:
Realized that you may just want to defer the completion, not necessarily a threading concern?

Reply:
Maybe a Task.Yield() would be enough here.  The goal is simply to ensure the _tcs is completed asynchronously.

Reply:
Yield is probably better indeed.